loguru/_logger.py

```diff
@@ -1752,6 +1754,46 @@ def configure(self, *, handlers=None, levels=None, extra=None, patcher=None, act
 
         return [self.add(**params) for params in handlers]
 
+    def _replace_core(self, core: Core):
+        self._core = core
+
+    def reinstall(self):
+        """Reinstall the core of logger.
+
+        When using multiprocessing, you can pass logger as a parameter to the target of
+        ``multiprocessing.Process``, and run this method once, thus you don't need to pass logger to every
+        function you called in the same process with spawn multiprocessing.
+
+        Examples
+        --------
+        >>> def subworker(logger_):
+        ...     logger_.reinstall()
+        ...     logger.info("Child")
+        ...     deeper_subworker()
+
+        >>> def deeper_subworker():
+        ...     logger.info("Grandchild")
+
+        >>> def test_process_spawn():
+        ...     spawn_context = multiprocessing.get_context("spawn")
+        ...     logger.add("file.log", context=spawn_context, enqueue=True, catch=False)
+        ...
+        ...     process = spawn_context.Process(target=subworker, args=(logger,))
+        ...     process.start()
+        ...     process.join()
+
+        ...     assert process.exitcode == 0
+
+        ...     logger.info("Main")
+        ...     logger.remove()
+        """
+        if self._own_pid == os.getpid():  # same process
+            return
+        from loguru import logger
+
+        logger._replace_core(self._core)
```

Comment:
I think we can remove the `_replace_core()` and do `logger._core = self._core` directly.

Reply:
Actually I also like `logger._core = self._core` directly, but I saw your comment in [#818](https://github.com/Delgan/loguru/issues/818#issuecomment-1538688196) so I thought you prefer this way XD. I'll change it.